### PR TITLE
Update _eval_integral, making it handle limit case where the exponent's coefficient becomes zero

### LINKED ISSUE #27280: Sympy gives a wrong result for an integral containing complex constants
I asked SymPy to evaluate the following integral:

$$
\int\limits_{0}^{1} \left(b-a\right) e^{t (re{\left(b\right))} + \left(1 - t\right) (re{\left(a\right))}} dt,
$$

with $re(a)$ and $re(b)$ being the real parts of two complex constants $a$ and $b$. If $re(a) = re(b)$, the integral should give $(b-a)e^{re(a)}$.

I use the following code to calculate the integral:

```
# SymPy Version is 1.13.2

import sympy as sp

a, b, z = sp.symbols('a, b, z')
t = sp.symbols('t', real=True)

f = (b - a) * sp.exp(t * sp.re(b) + (1 - t) * sp.re(a))

sp.integrate(f, (t, 0, 1))
```

and the result is

$$
\begin{cases} - \frac{\left(a - b\right) e^{{re}{\left(a\right)}}}{{re}{\left(a\right)} - {re}{\left(b\right)}} + \frac{\left(a - b\right) e^{{re}{\left(b\right)}}}{{re}{\left(a\right)} - {re}{\left(b\right)}} & \text{for} & {re}{\left(a\right)} - {re}{\left(b\right)} \neq 0 \\ & \text{and} & (b-a) & \text{otherwise} \end{cases}
$$

The SymPy result for the $re(a)\neq re(b)$ case is correct but for the other case, a factor of $e^{re(a)}$ is missing.

### PR DESCRIPTION
**Pull request by zelosleone**
#### References to other Issues or PRs
Fixes #27280 

#### Brief description of what is fixed or changed
I think we need to update _eval_integral function for this. The issue seems like comes from definition not correctly handling the limit case where the exponent's coefficient becomes zero.  My solution involves adding a Piecewise condition for this, re(a) == re(b) will be special case, and if it's not it's going to use the general method.

#### Release Notes

* integrals
  * changed the limit case handling

### CODE REVIEW

**Comment by Jatinbhardwaj-093** [comment]
Hi, I’ve actually been working on this problem for quite a while. I saw your solution, and it seems like a very generalized approach. Could you explain the intuition behind this solution?

File: sympy/integrals/integrals.py [hunk 1]
```diff
@@ -1160,8 +1159,50 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
                 parts.append(coeff * h)
             else:
                 return None
-
-        return Add(*parts)
+            current_expr = f
+            for lim in self.limits:
+                if len(lim) == 3:
```
Comment (oscarbenjamin) [inline sympy/integrals/integrals.py:1164]: This function is already far too long and messy. I would rather not extend it like this.

Instead there can be a new function for this.

File: sympy/integrals/integrals.py [hunk 2]
```diff
@@ -1160,8 +1159,50 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
...
+                else:
+                    var = lim[0]
+                    a = b = None
+                    is_definite = False
+                if is_definite and a is not None and b is not None:
+                    attempted = integrate(current_expr, (var, a, b), meijerg=meijerg, risch=risch,
+                                       manual=manual, heurisch=heurisch, conds=conds)
+                    if attempted == current_expr:
+                        break
+                    current_expr = attempted
```
Comment (oscarbenjamin) [inline sympy/integrals/integrals.py:1180]: What is this doing? Hasn't the code above already attempted to evaluate the integral? Why is this recursively calling `integrate`?

File: sympy/integrals/integrals.py [hunk 3]
```diff
@@ -1160,8 +1159,50 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
...
+                else:
+                    if is_definite and (a is None or b is None):
+                        A = a if a is not None else -oo
+                        B = b if b is not None else oo
+                        attempted = integrate(current_expr, (var, A, B),
```
Comment (oscarbenjamin) [inline sympy/integrals/integrals.py:1185]: This should not arbitrarily use infinity for the limits if no limits are given.

**Pull request closed without merging**